### ISSUE TITLE
Create diagrams for various processes in the NuCypher Network

### ISSUE THREAD

**Issue by derekpierre** [nucypher/nucypher#1347]
These could probably be used within docs and learning portal (? - #1001).

[ ] Alice.grant - How Alice communicates with Ursula and the blockchain for policies
[ ] Enrico.encrypt - How Alice and Enrico communicate, and Enrico's use of storage (capsule)
[ ] Bob.retrieve - How Bob communicates with Ursula for re-encryptions
[ ] Ursula.reencrypt - Receive kFrag from Alice, receive capsule and produce cFrag
[ ] Alice.revoke (maybe?) - might be for a separate PR - some details still being worked through
[ ] Contracts (via #1112)

**Comment by derekpierre** [comment]
@cygnusv / @arjunhassard / @mswilkison interested to get your thoughts here.

I initially filed this issue because of some questions we were getting in Discord. I started looking to match up spots for diagrams with content the docs. I created an initial overview diagram  (https://github.com/derekpierre/nucypher/blob/docs_diagrams/docs/source/.static/img/nucypher_overview.svg) that is used on the index page of our docs as part of the commit - https://github.com/derekpierre/nucypher/commit/fcc88feed9650d9b2502fd93d5563f640725feaa. 

However, we don't really have any good spots in the docs currently to add more in-depth detailed diagrams for processes like Alice.grant (Ursulas and Blockchain), or Bob.retrieve (Storage and Ursula), ... Of course, I can create a specific section in the docs for character interactions for the content but I'm trying to determine if it is best to have those diagrams within the docs itself, or whether they are better served in the learning portal (or perhaps both). Is it too much detail for the docs? The docs generally feel more like dev-docs to me, and the process diagrams may be more appropriate in the learning portal - that being said, we all know how hesitant people can be to access learning portals.

wdyt?
